services/src/main/java/io/druid/cli/CliInternalHadoopIndexer.java

```diff
@@ -116,7 +117,14 @@ public HadoopDruidIndexerConfig getHadoopDruidIndexerConfig()
         if (argumentSpec.startsWith("{")) {
           config = HadoopDruidIndexerConfig.fromString(argumentSpec);
         } else {
-          config = HadoopDruidIndexerConfig.fromFile(new File(argumentSpec));
+          URI argumentSpecUri = new URI(argumentSpec);
```

Comment:
are we sure argumentSpec is always a valid URI, since before it was just a simple path?
sometimes paths containing . or .. are not valid URIs.